File: apps/desktop/src-tauri/src/api/projects.rs
use crate::api::ApiClient;
use crate::error::AppResult;
use crate::models::ProjectInfo;

pub async fn list_projects(client: &ApiClient) -> AppResult<Vec<ProjectInfo>> {
    client.get("/projects").await
}

pub async fn add_project(client: &ApiClient, path: &str) -> AppResult<ProjectInfo> {
    let body = serde_json::json!({ "path": path });
    client.post("/projects", &body).await
}

File: apps/console-macos/Sources/ConsoleCore/Api/Projects.swift
import Foundation

// Mirrors api/projects.rs

public extension ApiClient {
    func listProjects() async throws -> [ProjectInfo] {
        try await get("/projects")
    }

    func addProject(path: String) async throws -> ProjectInfo {
        struct Body: Encodable, Sendable { let path: String }
        return try await post("/projects", body: Body(path: path))
    }
}
